railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -251,6 +258,28 @@ def create_vendor_files
         build(:vendor)
       end
 
+      def delete_app_assets_if_api_option
+        if options[:api]
+          remove_dir 'app/assets'
+          remove_dir 'lib/assets'
+          remove_dir 'tmp/cache/assets'
+          remove_dir 'vendor/assets'
+        end
+      end
+
+      def delete_app_helpers_if_api_option
+        if options[:api]
+          remove_dir 'app/helpers'
+          remove_dir 'test/helpers'
+        end
+      end
+
+      def delete_app_views_if_api_option
+        if options[:api]
+          remove_dir 'app/views'
+        end
+      end
```

Comment:
app/views is generated by jbuilder itself when you run the generators for the first time.